core-services/chat/src/main/java/org/silverpeas/core/chat/servers/OpenfireServer.java

```diff
@@ -0,0 +1,178 @@
+package org.silverpeas.core.chat.servers;
+
+import org.silverpeas.core.SilverpeasExceptionMessages;
+import org.silverpeas.core.admin.user.model.User;
+import org.silverpeas.core.chat.ChatServerException;
+import org.silverpeas.core.chat.ChatUser;
+import org.silverpeas.core.util.SettingBundle;
+import org.silverpeas.core.util.StringUtil;
+import org.silverpeas.core.util.logging.SilverLogger;
+
+import javax.ws.rs.core.Response;
+
+/**
+ * <p>Openfire server management service</p>
+ * <p>It implements the access mechanism to the XMPP server OpenFire.</p>
+ * @author remipassmoilesel
+ */
+public class OpenfireServer implements ChatServer {
+
+  private SilverLogger logger = SilverLogger.getLogger(this);
+
+  /**
+   * URL target of the REST requests.
+   */
+  private final String url;
+
+  private final HttpRequester requester;
+
+  public OpenfireServer() {
+    SettingBundle settings = ChatServer.getChatSettings();
+    this.url = settings.getString("chat.xmpp.restUrl") + "/users";
+    String key = settings.getString("chat.xmpp.restKey");
+    this.requester = new HttpRequester(key);
+  }
+
+  @Override
+  public void createUser(final User user) throws ChatServerException {
+    // get cross domain login and password
+    ChatUser chatUser = ChatUser.fromUser(user);
+    Response response = null;
+    try {
+      HttpRequester requester = request(url);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "requester" which hides the field declared at line 27. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AHiddenFieldCheck)